### DISCUSSION TITLE
Mac M1 - can't install elasticsearch@6

### DISCUSSION THREAD

**Discussion by mksada** [Homebrew/discussions#925] [Getting started]
Please help with my issue.

I'm trying to install elasticsearch@6 to my macbook with M1 processor.

At this documentation it is said that Apple Silicon is supported: https://formulae.brew.sh/formula/elasticsearch@6

When I hit brew install elasticsearch@6 I got this log:

`Error: elasticsearch@6: no bottle available!
You can try to install from source with:
  brew install --build-from-source elasticsearch@6
Please note building from source is unsupported. You will encounter build
failures with some formulae. If you experience any issues please create pull
requests instead of asking for help on Homebrew's GitHub, Twitter or any other
official channels.` 

### Here is my brew doctor log:

`Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /usr/local/bin/pkg-config
  /usr/local/bin/jemalloc-config
  /usr/local/bin/freetype-config
  /usr/local/bin/libpng16-config
  /usr/local/bin/libpng-config
  /usr/local/bin/pspell-config
  /usr/local/bin/python3.9-config
  /usr/local/bin/python3-config
  /usr/local/bin/pcre2-config
  /usr/local/bin/pcre-config
  /usr/local/bin/php-config

Warning: gettext files detected at a system prefix.
These files can cause compilation and link failures, especially if they
are compiled with improper architectures. Consider removing these files:
  /opt/homebrew/lib/libgettextlib.dylib
  /opt/homebrew/lib/libintl.dylib
  /opt/homebrew/include/libintl.h
  /usr/local/lib/libgettextlib.dylib
  /usr/local/lib/libintl.dylib
  /usr/local/include/libintl.h

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libevent-2.1.7.dylib
  /usr/local/lib/libevent_core-2.1.7.dylib
  /usr/local/lib/libevent_extra-2.1.7.dylib
  /usr/local/lib/libevent_pthreads-2.1.7.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/evdns.h
  /usr/local/include/event.h
  /usr/local/include/evhttp.h
  /usr/local/include/evrpc.h
  /usr/local/include/evutil.h

Warning: Unbrewed '.la' files were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.la' files:
  /usr/local/lib/libevent.la
  /usr/local/lib/libevent_core.la
  /usr/local/lib/libevent_extra.la
  /usr/local/lib/libevent_pthreads.la

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/libevent.pc
  /usr/local/lib/pkgconfig/libevent_core.pc
  /usr/local/lib/pkgconfig/libevent_extra.pc
  /usr/local/lib/pkgconfig/libevent_pthreads.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libevent.a
  /usr/local/lib/libevent_core.a
  /usr/local/lib/libevent_extra.a
  /usr/local/lib/libevent_pthreads.a`

### and brew config log:
`HOMEBREW_VERSION: 3.0.2

ORIGIN: https://github.com/Homebrew/brew
HEAD: b712bcc01e8fdaae25d6f232f51637bf00db307f
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 97e9d1f96b976d74aa0736fac65aa7ad890035b9
Core tap last commit: 73 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.24.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.1-arm64
CLT: 12.4.0.0.1.1610135815
Xcode: 12.4
Rosetta 2: false`

**Comment by SMillerDev** [comment]
It's not actually supported, that's why "apple silicon" is empty

**Comment by carlocab** [comment]
For reference, here's what that box should look like for a formula that does have an ARM bottle: https://formulae.brew.sh/formula/hello#default

**Comment by joshuapinter** [comment]
Trying to install Elasticsearch on a new M1 Mac Mini and ran into the same problem. However, tapping Elasticsearch's "keg" and installing it from there seemed to work no problem: 

```bash
brew tap elastic/tap

brew install elastic/tap/elasticsearch-full
```

From their docs: https://www.elastic.co/guide/en/elasticsearch/reference/current/brew.html

**Reply by alexhayes** [reply]
While this does work, it installs ElasticSearch 7 - not 6.

**Reply by joshuapinter** [reply]
Yup, you're right. Didn't notice that the first time around.

**Reply by dlagez** [reply]
good suggestion

**Reply by dorianmarie** [reply]
I'm getting:

```
Error: elastic/tap/elasticsearch-full: undefined method `plist_options' for Formulary::FormulaNamespacebf9a1382d41b546c07ca9fb03a7076777de36a194bc5d2ee6351261ec59a8ed3::ElasticsearchFull:Class
```

**Comment by alex-ross** [comment]
@mksada hopefully you already solved your problem. But we did need to install an older version of Elasticsearch for a projekt. Created a formula for it. It's version 5.6 but may help you or somebody.

https://github.com/standout/homebrew-tap#elasticsearch56

**Comment by thomasmokuaomweri2011** [comment]
The official elasticsearch is not supported yet on m1. However if on docker there are some images which can work eg `webhippie/elasticsearch:6.4` in https://hub.docker.com/r/webhippie/elasticsearch
